src/passes/hash-stringify-walker.cpp

```diff
@@ -147,6 +148,45 @@ std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::dedupe(
   return result;
 }
 
+std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::removeOverlaps(
+  const std::vector<SuffixTree::RepeatedSubstring>& substrings) {
+  std::vector<Interval> intervals;
+  std::unordered_map<Interval, unsigned> intervalMap;
+
+  // Construct intervals
+  for (Index i = 0; i < substrings.size(); i++) {
+    auto substring = substrings[i];
+    for (auto startIdx : substring.StartIndices) {
+      auto interval =
+        Interval(startIdx,
+                 startIdx + substring.Length - 1,
```

Comment:
I'm surprised we're using intervals inclusive of their ends. Would this work without the `- 1` as well?